windows/cross-compile/build.sh

```diff
@@ -92,1399 +76,45 @@ then
-" > toolchain.cmake
-
-# Run tests using Wine
-set +u
-if [[ -n "$TRAVIS_CI_STAGE" ]]
-then
-  echo "SET(TEST_CROSSCOMPILING_EMULATOR /usr/bin/wine)" >> toolchain.cmake
-fi
-set -u
-
 # Spell check on windows currently not supported, disable
 if [[ "$BUILD_TYPE" == "release" ]]
 then
-  cmake -DCMAKE_TOOLCHAIN_FILE=./toolchain.cmake \
+  cmake -DCMAKE_TOOLCHAIN_FILE=/build/windows-toolchain.cmake \
+        -DCMAKE_PREFIX_PATH=/windows \
         -DCMAKE_BUILD_TYPE=Release \
         -DSPELL_CHECK=OFF \
         -DDESKTOP_NOTIFICATIONS=ON \
         -DUPDATE_CHECK=ON \
         -DSTRICT_OPTIONS=ON \
-        ..
+        "$QTOX_SRC_DIR"
 elif [[ "$BUILD_TYPE" == "debug" ]]
 then
-  cmake -DCMAKE_TOOLCHAIN_FILE=./toolchain.cmake \
+  cmake -DCMAKE_TOOLCHAIN_FILE=/build/windows-toolchain.cmake \
+        -DCMAKE_PREFIX_PATH=/windows \
         -DCMAKE_BUILD_TYPE=Debug \
         -DSPELL_CHECK=OFF \
         -DDESKTOP_NOTIFICATIONS=ON \
         -DUPDATE_CHECK=ON \
         -DSTRICT_OPTIONS=ON \
-        ..
+        "$QTOX_SRC_DIR"
 fi
 
-make
+make -j $(nproc)
 
+mkdir -p "${QTOX_PREFIX_DIR}"
 cp qtox.exe $QTOX_PREFIX_DIR
-cp $QT_PREFIX_DIR/bin/Qt5Core.dll \
-   $QT_PREFIX_DIR/bin/Qt5Gui.dll \
-   $QT_PREFIX_DIR/bin/Qt5Network.dll \
-   $QT_PREFIX_DIR/bin/Qt5Svg.dll \
-   $QT_PREFIX_DIR/bin/Qt5Widgets.dll \
-   $QT_PREFIX_DIR/bin/Qt5Xml.dll \
-   $QTOX_PREFIX_DIR
-cp -r $QT_PREFIX_DIR/plugins/imageformats \
-      $QT_PREFIX_DIR/plugins/platforms \
-      $QT_PREFIX_DIR/plugins/iconengines \
-      $QTOX_PREFIX_DIR
-cp {$OPENSSL_PREFIX_DIR,$SQLCIPHER_PREFIX_DIR,$FFMPEG_PREFIX_DIR,$OPENAL_PREFIX_DIR,$QRENCODE_PREFIX_DIR,$EXIF_PREFIX_DIR,$OPUS_PREFIX_DIR,$SODIUM_PREFIX_DIR,$VPX_PREFIX_DIR,$TOXCORE_PREFIX_DIR}/bin/*.dll $QTOX_PREFIX_DIR
-
-cp "$SNORE_PREFIX_DIR/bin/libsnore-qt5.dll" $QTOX_PREFIX_DIR
-mkdir -p "$QTOX_PREFIX_DIR/libsnore-qt5"
-cp "$SNORE_PREFIX_DIR/lib/plugins/libsnore-qt5/libsnore_backend_windowstoast.dll" "$QTOX_PREFIX_DIR/libsnore-qt5"
-cp "$SNORE_PREFIX_DIR/bin/SnoreToast.exe" $QTOX_PREFIX_DIR
-
-cp /usr/lib/gcc/$ARCH-w64-mingw32/*-posix/{libgcc_s_*.dll,libstdc++*.dll,libssp*.dll} $QTOX_PREFIX_DIR
-cp /usr/$ARCH-w64-mingw32/lib/libwinpthread*.dll $QTOX_PREFIX_DIR
-
-# Setup wine
-# Note that SQLCipher and FFmpeg (maybe more?) seem to setup ~/.wine on their
-# own, but to the wrong bitness (always 64-bit?), when we want a matching
-# bitness here for mingw-ldd, so remove it before proceeding.
-rm -rf ~/.wine
-if [[ "$ARCH" == "i686" ]]
-then
-  export WINEARCH=win32
-elif [[ "$ARCH" == "x86_64" ]]
-then
-  export WINEARCH=win64
-fi
-winecfg
+cp -r /export/* $QTOX_PREFIX_DIR
 
 # Run tests (only on Travis)
 set +u
 if [[ -n "$TRAVIS_CI_STAGE" ]]
 then
-  # Add libgcc_s_*.dll, libwinpthread-1.dll, QtTest.dll, etc. into PATH env var of wine
-  export WINEPATH=`cd $QTOX_PREFIX_DIR ; winepath -w $(pwd)`\;`winepath -w $QT_PREFIX_DIR/bin/`
+  export WINEPATH='/export;/windows/bin'
   export CTEST_OUTPUT_ON_FAILURE=1
   make test
```

Comment:
For tests to work, `TEST_CROSSCOMPILING_EMULATOR` needs to be set, but you have removed it.

Reply:
Yeah this was a mistake. I had already done the work of understanding why this worked fine on my machine and not on CI (if your host has binfmt installed you don't need this, but otherwise you do). I'm not sure how I ended up in a state where these tests were no longer being run, since I clearly was running them at some point to do that investigation.

Thanks for commenting here, this is a massive mistake